File: src/java/pe_0004.java
public class pe_0004 {
    public static int reverse_number(int n) {
        int m = 0;
        while (n != 0) {
            m *= 10;
            m += n % 10;
            n /= 10;
        }
        return m;
    }
    
    public static boolean is_palindromic(int n) {
        return n == reverse_number(n);
    }

    public static void main(String[] args) {
        int max = 0;

        for (int a = 1; a < 1000; a += 1) {
            for (int b = a; b < 1000; b += 1) {
                int p = a * b;
                if (is_palindromic(p) && max < p) {
                    max = p;
                }
            }
        }
    
        System.out.println(max);   
    }
}

File: src/jl/pe_0004.jl
function reverse_number(n::Integer)
    m = 0
    while (n != 0)
        m *= 10
        m += n % 10
        n ÷= 10
    end
    return m
end

function is_palindromic(n)
    return n == reverse_number(n)
end

function main()
    max = 0

    for a in 100:1000
        for b in a:1000
            p = a * b
            if (is_palindromic(p) && max < p)
                max = p
            end
        end
    end

    println(max)
end

main()
